clients/fides-js/src/components/Overlay.tsx

```diff
@@ -53,15 +71,20 @@ const Overlay: FunctionComponent<OverlayProps> = ({
       // Update modal link to trigger modal on click
       const modalLink = modalLinkEl;
       modalLink.onclick = () => {
-        setModalIsOpen(true);
+        handleOpenModal();
         setBannerIsOpen(false);
       };
       // Update to show the pre-existing modal link in the DOM
       modalLink.classList.add("fides-modal-link-shown");
     } else {
       debugLog(options.debug, "Modal link element not found.");
     }
-  }, [modalLinkEl, options.debug]);
+  }, [modalLinkEl, options.debug, handleOpenModal]);
+
+  const showBanner = useMemo(
+    () => experience.show_banner && hasActionNeededNotices(experience),
+    [experience]
+  );
```

Comment:
❗ this is an unrelated change, but for some reason the tests only failed once I was on this branch, even though my changes were unrelated. the test that failed was 

https://github.com/ethyca/fides/blob/e744e34b959a9949c00bd5c806b70d38e510ade5/clients/privacy-center/cypress/e2e/consent-banner.cy.ts#L883-L885

for the use case where the user already has `current_preference`s for every notice. the test would fail and the banner would still show up, even though all notices had been opted into.